merlin/models/torch/transforms/agg.py

```diff
@@ -0,0 +1,193 @@
+from typing import Dict, Union
+
+import torch
+from torch import nn
+
+from merlin.models.torch.registry import registry
+
+
+@registry.register("concat")
+class Concat(nn.Module):
+    """Concatenate tensors along a specified dimension.
+
+    Parameters
+    ----------
+    dim : int
+        The dimension along which the tensors will be concatenated.
+        Default is -1.
+
+    Examples
+    --------
+    >>> concat = Concat()
+    >>> feature1 = torch.tensor([[1, 2], [3, 4]])  # Shape: [batch_size, feature_dim]
+    >>> feature2 = torch.tensor([[5, 6], [7, 8]])  # Shape: [batch_size, feature_dim]
+    >>> input_dict = {"feature1": feature1, "feature2": feature2}
+    >>> output = concat(input_dict)
+    >>> print(output)
+    tensor([[1, 2, 5, 6],
+             [3, 4, 7, 8]])  # Shape: [batch_size, feature_dim*number_of_features]
+
+    """
+
+    def __init__(self, dim: int = -1):
+        super().__init__()
+        self.dim = dim
+
+    def forward(self, inputs: Dict[str, torch.Tensor]) -> torch.Tensor:
+        """
+        Concatenates input tensors along the specified dimension.
+
+        Parameters
+        ----------
+        inputs : Dict[str, torch.Tensor]
+            A dictionary where keys are the names of the tensors
+            and values are the tensors to be concatenated.
+
+        Returns
+        -------
+        torch.Tensor
+            A tensor that is the result of concatenating
+            the input tensors along the specified dimension.
+
+        Raises
+        ------
+        RuntimeError
+            If the input tensor shapes don't match for concatenation
+            along the specified dimension.
+        """
+        sorted_tensors = [inputs[name] for name in sorted(inputs.keys())]
```

Comment:
Sorting the dictionary by keys makes sense from an implementation perspective, but I wonder if it would be confusing from the user perspective, given that Python dictionaries are insertion ordered. Not sure what's best, maybe just make this behavior clear in the docstring?

Reply:
Yeah, I agree we should add it to the doc-string. I think in most cases it will be insertion order of the data-loader which is not user-code so ordering alphabetically at least leads to consistent behaviour.